nltk/tree/prettyprinter.py

```diff
@@ -75,7 +75,7 @@ def __init__(self, tree, sentence=None, highlight=()):
             leaves = tree.leaves()
             if (
                 leaves
-                and not any(len(a) == 0 for a in tree.subtrees())
+                and any(len(a) > 0 for a in tree.subtrees())
```

Comment:
The replacement contains a logical error. NB:

```
>>> not any(len(a) == 0 for a in [[1], []])
False
>>> any(len(a) > 0 for a in [[1], []])
True
```

Reply:
Yeah right, my bad. 

I believe it should be replaced with all instead of any.

Considering the below test cases:- 
```
>>> not any(len(a) == 0 for a in [[1], []])
False
>>> all(len(a) > 0 for a in [[1], []])
False

>>> not any(len(a) == 0 for a in [[], []])
False
>>> all(len(a) > 0 for a in [[], []])
False

>>> not any(len(a) == 0 for a in [[1], [1]])
True
>>> all(len(a) > 0 for a in [[1], [1]])
True
```